modules/Emails/EmailsController.php

```diff
@@ -139,19 +139,43 @@ public function action_ComposeView()
      */
     public function action_send()
     {
-        $this->bean = $this->bean->populateBeanFromRequest($this->bean, $_REQUEST);
-        $this->bean->save();
+        global $current_user;
+        global $app_strings;
 
-        $this->bean->handleMultipleFileAttachments();
+        $this->bean = $this->bean->populateBeanFromRequest($this->bean);
+        $inboundEmailAccount = new InboundEmail();
+        $inboundEmailAccount->retrieve($_REQUEST['inbound_email_id']);
 
-        if ($this->bean->send()) {
-            $this->bean->status = 'sent';
+        if($this->userIsAllowedToSendEmail($current_user, $inboundEmailAccount, $this->bean)) {
             $this->bean->save();
+
+            $this->bean->handleMultipleFileAttachments();
+
+            if ($this->bean->send()) {
+                $this->bean->status = 'sent';
+                $this->bean->save();
```

Comment:
Shouldn't the status also be saved when it failed to send?
I know you didn't touch this part, but it is interesting nonetheless.